gcp/api/server.py

```diff
@@ -223,13 +224,17 @@ def _is_version_affected(affected_packages,
                          version,
                          normalize=False):
   """Returns whether or not the given version is within an affected ECOSYSTEM
-  range."""
+  range.
+  """
   for affected_package in affected_packages:
     if package_name and package_name != affected_package.package.name:
       continue
 
-    if ecosystem and ecosystem != affected_package.package.ecosystem:
-      continue
+    if ecosystem:
+      # If package ecosystem has a :, also try ignoring parts after it.
+      if affected_package.package.ecosystem != ecosystem and\
```

Comment:
nit: prefer ( ) for line continuations per https://google.github.io/styleguide/pyguide.html#32-line-length

Also can we just do:

```python
if (ecosystems.normalize(affected_package.package.ecosystem) !=
    ecosystems.normalize(ecosystem)): 
  continue
```

This makes the logic a bit easier to follow.

Reply:
Yes, but that behavior will also be different, if someone is searching specifically for Debian:4.0 for example, that will show packages with Debian:11. With the current implementation, only if they search for the general Debian ecosystem, will it show all debian entries

Reply:
Thanks for clarifying. That makes sense. Let's keep your approach then.

Can you add some integration tests for this?